src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -297,7 +300,7 @@ public ReadOnlySpan<T> Peek(int maxCount)
             }
 
             // Not enough contiguous Ts, allocate and copy what we can get
-            return PeekSlow(new T[maxCount]);
+            return PeekSlow(_arrayPool.Rent(maxCount));
```

Comment:
I don't see any calls to `.Return` in this PR.  Where is the array being returned to the pool?

Reply:
+1

Reply:
It isn't being returned as it isn't actually renting in the default implementation and isn't always renting here anyway. See my other comments.